### PR TITLE
Changes for TanninOne/modorganizer#309 (2 of 3)

### PR DESCRIPTION
**Pull request by ThosRTanner**
Changes for TanninOne/modorganizer#309

This makes checkfnis plugin detect if it is running itself and behave appropriately.

### CODE REVIEW

File: src/checkfnis.cpp [hunk 1]
```diff
@@ -129,8 +156,39 @@ QString CheckFNIS::generateIdentifier() const
...
+}
+
 bool CheckFNIS::fnisCheck(const QString &application)
 {
+  if (!m_MOInfo->pluginSetting(name(), "enabled").toBool()) {
+    return true;
+  }
+
+  QString fnisApp;
+  if (appIsFNIS(application, &fnisApp)) {
```
Comment (TanninOne) [inline src/checkfnis.cpp]: appIsFNIS returns false if fnis isn't installed so there is no early exit in this case and you will get a nasty error message further down.

File: src/checkfnis.cpp [hunk 2]
```diff
@@ -1,29 +1,44 @@
 #include "checkfnis.h"
 
 #include "iplugingame.h"
+#include "pluginsetting.h"
 #include "report.h"
-#include <scopeguard.h>
```
Comment (TanninOne) [inline src/checkfnis.cpp:5]: This isn't right? scopeguard.h and questionboxmemory.h aren't part of this repository.

Comment (ThosRTanner) [inline src/checkfnis.cpp:5]: I wasn't sure of the rules. scopeguard.h and questionboxmemory.h are both part of mod organizer, but not in the checkfnis repository, but the same applies to iplugingame.h and report.h

File: src/checkfnis.cpp (hunk 1 above)
Comment (ThosRTanner) [inline src/checkfnis.cpp]: oops. fixing

**Pull request merged**